### PR TITLE
Enabled shared library building on Linux.

### PR DESCRIPTION
**Pull request by xvzcf**
Making this a draft until I add CI tests.

### CODE REVIEW

File: fuzz/oids.txt [hunk 1]
```diff
@@ -1063,3 +1063,22 @@ OBJ_id_tc26_gost_3410_2012_256_paramSetC="\x2A\x85\x03\x07\x01\x02\x01\x01\x03"
 OBJ_id_tc26_gost_3410_2012_256_paramSetD="\x2A\x85\x03\x07\x01\x02\x01\x01\x04"
 OBJ_hmacWithSHA512_224="\x2A\x86\x48\x86\xF7\x0D\x02\x0C"
 OBJ_hmacWithSHA512_256="\x2A\x86\x48\x86\xF7\x0D\x02\x0D"
+OBJ_oqsdefault="\x2B\xCE\x0F\x01\x01"
```
Comment (dstebila) [inline fuzz/oids.txt:1066]: Do these need to be generated from templates? Or are we just putting a few in here as potentially tests but not necessarily the whole suite?

File: util/libcrypto.num [hunk 2]
```diff
@@ -1019,7 +1019,7 @@ ASN1_UTCTIME_it                         1013	1_1_0	EXIST:!EXPORT_VAR_AS_FUNCTION
 ASN1_UTCTIME_it                         1013	1_1_0	EXIST:EXPORT_VAR_AS_FUNCTION:FUNCTION:
 i2d_DSA_PUBKEY_fp                       1014	1_1_0	EXIST::FUNCTION:DSA,STDIO
 X509at_get_attr_by_OBJ                  1015	1_1_0	EXIST::FUNCTION:
-EVP_MD_CTX_copy_ex                      1016	1_1_0	EXIST::FUNCTION:
+EVP_MD_CTX_copy_ex                      1016	1_1_0	NOEXIST::FUNCTION:
```
Comment (dstebila) [inline util/libcrypto.num]: Is this related to some change we made?  Or a bug in upstream?

File: fuzz/oids.txt (hunk 1 above)
Comment (xvzcf) [reply fuzz/oids.txt:1066]: This file is autogenerated.

File: util/libcrypto.num (hunk 2 above)
Comment (xvzcf) [reply util/libcrypto.num]: I've fixed this and added a clarifying comment to the source.

**Comment by dstebila** [comment]
What do we need to do to finish this off?

**Comment by xvzcf** [comment]
Once [this](https://github.com/open-quantum-safe/openssl/pull/119) is merged in, I will add a template file that removes all the BIKE kems from the OpenSSL source only for the Debian shared build, which should hopefully do the trick.

**Comment by christianpaquin** [comment]
Looks like the builds fail because they are looking for the old qtesla algs. Does the code need to be updated?

**Comment by xvzcf** [comment]
Yes, I'm working on it right now.

**Comment by xvzcf** [comment]
It turns out removing BIKE through templating is tricker than expected. I think we'll have to keep [this issue](https://github.com/open-quantum-safe/openssl/issues/116) open, and for now I'll be removing the debian-shared build job.

File: fuzz/oids.txt [hunk 3]
```diff
@@ -1063,3 +1063,22 @@ OBJ_id_tc26_gost_3410_2012_256_paramSetC="\x2A\x85\x03\x07\x01\x02\x01\x01\x03"
...
+OBJ_dilithium3="\x2B\xCE\x0F\x02\x04"
+OBJ_dilithium4="\x2B\xCE\x0F\x02\x05"
+OBJ_p384_dilithium4="\x2B\xCE\x0F\x02\x06"
+OBJ_picnicl1fs="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x01\x01"
+OBJ_p256_picnicl1fs="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x01\x02"
+OBJ_rsa3072_picnicl1fs="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x01\x03"
+OBJ_qteslai="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x02\x01"
+OBJ_p256_qteslai="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x02\x02"
+OBJ_rsa3072_qteslai="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x02\x03"
+OBJ_qteslaiiisize="\x2B\x06\x01\x04\x01\x82\x37\x59\x02\x02\x04"
```
Comment (christianpaquin) [inline fuzz/oids.txt]: These should be re-generated to use the new qTesla algs

**Review by christianpaquin — CHANGES_REQUESTED** [review]

**Review by christianpaquin — APPROVED** [review]

**Comment by christianpaquin** [comment]
It would be good to be more descriptive in the README.md on explaining how to build with shared libs. Is it really that complicated to explain?

**Comment by xvzcf** [comment]
It isn't, and I think a short note would do.

**Pull request merged**